Redux as a feature and code refactoring - adding env variables , removing console logs, array length check for dropdown , removing console logs

Shah-Arsalan opened the pull request:
Added redux for state management 
Refactored code :
Added env variables 
Created base URL
removed console.logs
Dropdown array length check

Shah-Arsalan committed:
refactor : dotenv , Base url, dropdown length check

Shah-Arsalan committed:
feat : added redux

manvithak commented on Components/CategoryNews.js:
```diff
@@ -0,0 +1,60 @@
+import Hyperlink from "react-native-hyperlink";
+import Icon from "react-native-vector-icons/MaterialCommunityIcons";
+
+const CategoryNews = ({ ele, id, setId }) => {
```
Give proper variable names. Write element instead of ele.

Shah-Arsalan replied:
addressed

manvithak commented on Components/Dropdown.js:
```diff
@@ -0,0 +1,108 @@
+        return (
+          <View style={styles.dropdownButtonStyle}>
+            <Text style={styles.dropdownButtonTxtStyle}>
+              {(selectedItem && selectedItem) || "Select category"}
```
I think this `selectedItem && selectedItem` is not necessary.

Shah-Arsalan replied:
addressed

manvithak commented on Components/Dropdown.js:
```diff
@@ -0,0 +1,108 @@
+    paddingVertical: 8,
+  },
+
+  dropdownItemTxtStyle: {
```
Please make dropdownItemTxtStyle to `dropdownItemTextStyle`

Shah-Arsalan replied:
addressed

manvithak commented on Redux/newsSlice.js:
```diff
@@ -0,0 +1,77 @@
+};
+
+export const getSources = createAsyncThunk(
+  "getSources",
```
Try to give action names as `GET_SOURCES`

Shah-Arsalan committed:
fix: address code review feedback

- Rename 'ele' to 'element' in CategoryNews
- Remove redundant selectedItem check- Standardize style naming to dropdownItemTextStyle
